Staking pool factory (#37)
near/initial-contracts: staking-pool-factory/README.md

# Staking Pool Factory Contract

> This contract allows to deploy new staking pool contract and automatically whitelist them.

Reviewer: ```suggestion
This contract deploys and automatically whitelists new staking pool contracts.
```

It allows any user to create an new whitelisted staking pool.

> The staking pool factory contract keeps the hardcoded binary of the staking pool contract within itself.

Reviewer: ```suggestion
The staking pool factory contract packages the binary of the staking pool contract within its own binary.
```

To create a new staking pool a user should issue a function call transaction and attach the required minimum deposit.
The entire deposit will be transferred to the newly created staking pool contract in order to cover the required storage.

> When a user issues a function call towards the factory to create a new staking pool the factory internally checks that

Reviewer: What does it mean to internally check?  Does this mean that it checks that the account ID doesn't exist as an account on chain?

> the staking pool account ID doesn't exists, validates arguments for the staking pool initialization and then issues a

Reviewer: nit
```suggestion
the staking pool account ID does not exists, validates arguments for the staking pool initialization and then issues a
```
